src/Tools/dotnet-monitor/DiagnosticPortValidateOptions.cs

```diff
@@ -12,10 +14,29 @@ internal class DiagnosticPortValidateOptions :
     {
         public ValidateOptionsResult Validate(string name, DiagnosticPortOptions options)
         {
+            var failures = new List<string>();
+
             if (options.ConnectionMode == DiagnosticPortConnectionMode.Listen
                 && string.IsNullOrEmpty(options.EndpointName))
             {
-                return ValidateOptionsResult.Fail(Strings.ErrorMessage_DiagnosticPortMissingInListenMode);
+                failures.Add(Strings.ErrorMessage_DiagnosticPortMissingInListenMode);
+            }
+
+            // On Windows, the server is implemented using Named Pipes.
+            // -1 means use as many as resources allow.
+            // Otherwise, the allowed instances are between 1 and 254.
+
+            // For Linux domain sockets, all values are valid with 0 representing 1 connection
+            // and other invalid values being normalized to a system defined maximum.
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) &&
+               ((options.MaxConnections < -1) || (options.MaxConnections == 0) || (options.MaxConnections > 254)))
```

Comment:
There should be a MaxConnections property on the server endpoint info source that you can use, which is an OS agnostic way of declaring to use the maximum number of connections allowed by the OS.

Also, you should need to check if it's Windows.